### PR TITLE
Add the list of contributors agreeing to LGPL licensing

### PR DESCRIPTION
**Pull request by marcusmueller**
* List of contributors, with short explanation
* added LGPLv3 license text as COPYING-LGPL
* moved GPLv3 license text to COPYING-GPL
* COPYING still contains the GPLv3, but with an explanation that new
  code is LGPL
* updated the contribution guide
* Removed the GPL license header from Readme, added explanation.

### CODE REVIEW

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (jdemel) [inline AUTHORS_GRANTING_LGPL_LICENSE.txt]: It should be "or later" in both cases. For GPL that should be the case anyways and for LGPL it should be too IMHO.

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (jdemel) [inline AUTHORS_GRANTING_LGPL_LICENSE.txt]: It should probably say "recontribute" or similar. According to the discussions we had in the corresponding GREP. @bhilburn  What's your insight on this?

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (jdemel) [inline AUTHORS_GRANTING_LGPL_LICENSE.txt]: I suggest to add email and github handle. These are the infos other projects gathered. Also, this would fix a few corner cases. I have a root <root@nXXXX...> in mind.

**Review by jdemel — COMMENTED** [review]
It looks good so far. Just had a few comments.

File: CONTRIBUTING.md
Comment (michaelld) [inline CONTRIBUTING.md]: Typo: "LPGPL" -> "LGPL"

File: README.md
Comment (michaelld) [inline README.md]: "their name" -> "their name, github handle, and email address used for VOLK commits -- possibly more than 1 entry if they submitted code from different handles and/or emails --" ... or something like this

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (michaelld) [reply AUTHORS_GRANTING_LGPL_LICENSE.txt]: actually in this case I think Marcus is correct: just GPLv3 and LGPLv3 ... but I'm open to arguments as to why one way or the other

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (michaelld) [inline AUTHORS_GRANTING_LGPL_LICENSE.txt]: "licensed differently (GPLv3)" -> "licensed differently than VOLK's current license (GPLv3)"

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (michaelld) [reply AUTHORS_GRANTING_LGPL_LICENSE.txt]: I'll admit I haven't seen this part of the GREP. @jdemel can you do a quick summary of where "recontribute" comes in there? It's really not obvious to me ... what's written looks fine (with minor corrections)!

**Review by michaelld — CHANGES_REQUESTED** [review]
good start!

**Comment by jdemel** [comment]
@michaelld from https://github.com/gnuradio/greps/pull/33/files around line 64

> ### All contributors need to re-submit their code under the new license
>
> Whether one submitted code under a CLA or with a DCO, every contributor must re-submit their code under the new license.
> A contributor may submit their code to any project, including VOLK under a new license, even if that code was submitted under a CLA.

I assume the correct phrasing would be "I re-submit my code ..."

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (jdemel) [reply AUTHORS_GRANTING_LGPL_LICENSE.txt]: I assume the interesting part in our license is 
https://github.com/gnuradio/volk/blob/237a6fc9242ea8c48d2bbd417a6ea14feaf7314a/COPYING#L563

Followed by our license statement:
https://github.com/gnuradio/volk/blob/main/README.md#license
> VOLK is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 3, or (at your option) any later version.

With these things in mind, I thought VOLK is currently licensed under GPLv3+ (or later) and I'd assume we want to re-license to LGPLv3+.

Finally, the [GREP](https://github.com/gnuradio/greps/pull/33/files) currently states:

> The proposed new license is [LGPLv3+](https://www.gnu.org/licenses/lgpl-3.0.html).

We should be consistent. I'd argue that's the most important part.

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (jdemel) [reply AUTHORS_GRANTING_LGPL_LICENSE.txt]: [Line 64: "All contributors need to re-submit their code under the new license"](https://github.com/gnuradio/greps/pull/33/files)

I assume the correct version would state: "contributors need to re-submit their code". And by adding their name to the list, they do so.

File: COPYING
Comment (jdemel) [inline COPYING]: LGPLv3+

File: COPYING
Comment (jdemel) [inline COPYING]: They should all state GPLv3+. I hope.

File: README.md
Comment (jdemel) [reply README.md]: I would suggest to update the license statement to LGPLv3+ and add your further explanations below. We can remove the explanations once the process is complete.

**Comment by marcusmueller** [comment]
> @michaelld from https://github.com/gnuradio/greps/pull/33/files around line 64
> 
> > ### All contributors need to re-submit their code under the new license
> > Whether one submitted code under a CLA or with a DCO, every contributor must re-submit their code under the new license.
> > A contributor may submit their code to any project, including VOLK under a new license, even if that code was submitted under a CLA.
> 
> I assume the correct phrasing would be "I re-submit my code ..."

Slightly confused, as on the call you said you did not want the code to be re-submitted, but instead have a list of authors who are willing to license as LGPL; so you explicitly did not want to start with a clean-slate LGPLv3 repo (or branch) and submit code into it as authors agree. In that case, no resubmission of code can happen, the code is already where it belongs...

So, I think I might have simply misunderstood you there. Do you want to either:

* Start from scratch and only submit code into a clean LGPLv3 codebase (resubmission approach)  xor
* Want to keep a list of authors that agreed to the new license and then flush out the code of which the authors were not willing to relicense?

This PR only makes sense in the second case. So, could you clarify?

File: COPYING
Comment (marcusmueller) [reply COPYING]: Well, that should be easy to check?

```
cd volk/kernels/volk
ag -L "GNU General" $(ag -l copyright)
```

yields a list of only two files that have a copyright statement, but aren't GPL, and these two happen to be already LGPL.

There's a bunch of kernel, orc and asm code that doesn't carry a proper copyright note (see attached 
[unlicensed.txt](https://github.com/gnuradio/volk/files/6759814/unlicensed.txt)), but this is the exhaustive list of their committers, found by `( for f in $(ag -L license) ; do; git log  --format="* %aN <%aE>" $f ; done ) |sort -u `: 

* Albin Stigo <albin.stigo@gmail.com>
* Doug Geiger <doug.geiger@bioradiation.net>
* ghostop14 <ghostop14@gmail.com>
* hcab14 <hcab14@gmail.com>
* Johannes Demel <demel@uni-bremen.de>
* Marcus Müller <mueller@kit.edu>
* Nathan West <nathan.west@nrl.navy.mil>
* Nathan West <nathan.west@okstate.edu>

I think we're safe when it comes to kernels.

**Comment by jdemel** [comment]
Yes. This is confusing. The way I understand @bhilburn is, we ask all contributors to re-submit their code under a new license. 
BUT: This is a legal discussion. Technically, they re-submit their code by adding their names to the list.

**Comment by michaelld** [comment]
Ditto what @jdemel wrote: "resubmit code" is being used as a legal term when relicensing that for our case will mean adding their name / info to the noted file. It does not mean literally submitting their code again in a PR for us to review (etc) but under the new license.

**Comment by michaelld** [comment]
I'm good with moving the license naming in the relicensing file to "LGPL3+" ... I think that's clear enough and precise enough

File: COPYING-LGPL
Comment (jdemel) [inline COPYING-LGPL]: Could you change this to `before version 3`? That'd be more verbose. At least for me.

File: AUTHORS_GRANTING_LGPL_LICENSE.txt
Comment (jdemel) [reply AUTHORS_GRANTING_LGPL_LICENSE.txt]: This is the license statement in most files.

https://github.com/gnuradio/volk/blob/237a6fc9242ea8c48d2bbd417a6ea14feaf7314a/kernels/volk/volk_32fc_s32fc_multiply_32fc.h#L7-L10

I suggest to use SPDX names.
[GNU General Public License v3.0 or later (GPL-3.0-or-later)](https://spdx.org/licenses/GPL-3.0-or-later.html)
[GNU Lesser General Public License v3.0 or later (LGPL-3.0-or-later)](https://spdx.org/licenses/LGPL-3.0-or-later.html)

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (mbr0wn) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]:
```suggestion
| 2021-08-25 | Martin Braun | mbr0wn        | martin@gnuradio.org, martin.braun@ettus.com  |
|            |                             |
```

**Review by mbr0wn — COMMENTED** [review]
We should not require people to update this file themselves -- if we have an email approval from them, or a github comment approval, we can update this file ourselves.

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (jdemel) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md:13]: As per @mbr0wn 's comment, we should add a sentence that tells people how to add themselves to this list.
```suggestion

## How to add myself to the table of authors
There are several options. Choose the one you prefer.
1. Open a PR and add yourself to the table.
2. Add a Github comment that clearly states your wish to be added to the table.
3. Write an email to the maintainers that clearly states your wish to be added to the table.

If you go with option 2 or 3, the maintainers add your statement to the corresponding git commit message for documentation purposes.
```

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (Ka-zam) [reply AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]: Add me:
| 2021-08-27 | Magnus Lundmark | Ka-zam | magnus@skysense.io |

**Review by Ka-zam — CHANGES_REQUESTED** [review]

**Comment by mbr0wn** [comment]
On the re-submit thing: Currently, VOLK2 is a project that is owned by the FSF and licensed GPLv3+. It's not owned by us, so we can't change a license. But we *can* create an entirely new project, VOLK3, that is owned by all, and LGPLv3+. Now, we'll still use the same repo and filenames etc., which also allows us to do `git blame`, `git bisect`, or whatever else we want across 2/3 boundaries. But in principle, we are creating an entirely new project with a new license and ownership. So I think re-submit is the right term, because contributor's can't give us permission the change the license of the existing project.

**Review by jdemel — APPROVED** [review]
LGTM.

Thanks @marcusmueller for starting this.

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (michaelld) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]: "is going to migrating" -> "is migrating" ... or something along those lines

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (michaelld) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]: "this gives the right" -> "this gives the VOLK project the right"

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (michaelld) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]: "maintainers add your" -> "maintainers will add your"

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (michaelld) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md:23]: duplicate sentence to the immediate above one ... choose one or the other

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (michaelld) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]: Please add mine:

` | 2021-09-04 | Michael Dickens | michaelld | mlk@alum.mit.edu, michael.dickens@ettus.com |`

File: COPYING-LGPL
Comment (michaelld) [inline COPYING-LGPL]: In the files `COPYING` and `README.md` we're using the terms `LGPL-3.0-or-later` and `GPL-3.0-or-later` whereas here we're using `LGPLv3` and `GPLv3`. This usage is inconsistent IMHO. I don't know which is the better / more correct way to go, but let's just be consistent ... but if I had to pick I'd use the terms from the other files.

File: README.md
Comment (michaelld) [inline README.md]: "are asked" -> "are required"

File: README.md
Comment (michaelld) [inline README.md]: "is moving" -> "is migrating" ... same as the file `AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md`

File: README.md
Comment (michaelld) [inline README.md]: `is still GPL-licensed,` -> `is still GPL-licensed.`

File: README.md
Comment (michaelld) [reply README.md]: `but new contributors` -> `New contributors`

File: README.md
Comment (michaelld) [inline README.md]: `to also allow LPGL-licensing by` -> `to relicense their GPL-3.0-or-later license code contributions to LGPL-3.0-or-later by`

**Review by michaelld — CHANGES_REQUESTED** [review]
Sorry for coming in late to the party ... better late than later?! Great start!

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (jdemel) [reply AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]:
```suggestion
VOLK is migrating from GNU General Public License v3.0 or later (GPL-3.0-or-later)
```

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (jdemel) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]:
```suggestion
this gives the VOLK project the right to use the current contributions under both LGPL-3.0-or-later and that other license.
```

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (jdemel) [reply AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]:
```suggestion
If you go with option 2 or 3, the maintainers will add your statement to the corresponding git commit message for documentation purposes.
```

Comment (jdemel) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]:
```suggestion
```

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (jdemel) [inline AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]:
```suggestion
| 2021-09-04 | Michael Dickens | michaelld | mlk@alum.mit.edu, michael.dickens@ettus.com |
|            |                             |
```

File: README.md
Comment (jdemel) [reply README.md]:
```suggestion
VOLK is migrating from the GNU General Public License v3.0 or later (GPL-3.0-or-later) 
```

File: README.md
Comment (jdemel) [reply README.md]:
```suggestion
At this point in time, much of the code in the repository is still GPL-licensed.
```

File: README.md
Comment (jdemel) [reply README.md]:
```suggestion
New contributors are required to use the LGPLv3+ for their code contributions. 
```

File: README.md
Comment (jdemel) [reply README.md]:
```suggestion
Existing contributors are very kindly requested to re-submit their GPL-3.0-or-later license code contributions to LGPL-3.0-or-later by
```
After various discussion, we ask everyone to re-submit their code. However, all contributors can re-submit by adding themselves to the list instead of actually opening a new PR with all their commits. That would be infeasible to handle.

**Comment by jdemel** [comment]
@michaelld I added suggestions to all of your comments. I really like that feature. It highlights the exact changes and one can go through, "add them to the batch" and just do a commit.
However, I'd separate your addition to the list into its own commit.

File: AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md
Comment (marcusmueller) [reply AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md]: please don't add more contributor lines to this PR: merge it first, I don't want to become the gatekeeper/bottleneck. Obviously, I'm adding Michael, but he'll be the last one.

**Comment by marcusmueller** [comment]
please don't add more contributor lines to this PR: merge it first, I don't want to become the gatekeeper/bottleneck. Obviously, I'm adding Michael, but he'll be the last one.

**Comment by marcusmueller** [comment]
@michaelld @mbr0wn 
Can you check whether your concerns have been addressed? Since @jdemel has already LGTM'ed, that would be the final blocker.

We need to merge this, as I will not be accepting any further PRs-on-PR to add re-submitters to the table anymore.

**Comment by marcusmueller** [comment]
@jdemel  as maintainer, you have push access to this branch. Please sign off on your own commits!

**Review by michaelld — APPROVED** [review]
LGMT ; thx for the quick changes per my prior review!

**Comment by michaelld** [comment]
@jdemel given the (slight) changes since your last review, I think it's worth the effort for you to re-review. Assuming you approve then please go ahead & merge ... and: Let's do this!!!

**Pull request merged**
